quiz-microsevice/business-logic/services/quiz-service.js

```diff
@@ -196,4 +196,37 @@ const createQuiz = async (
   return { ...quiz, questions: questionsWithoutQuizId };
 };
 
-module.exports = { createQuiz };
+const retrieveQuizzes = async (clientId) => {  
+  const { quizzes, questions } = await quizRepository.retrieveQuizzes(clientId);  
+
+  if (!Array.isArray(quizzes)) {  
+      throw new Error('Quizzes is not an array');  
+  }  
+  
+  const quizzesWithQuestions = quizzes.map(quiz => {  
```

Comment:
We don't need questions for each quiz for this endpoint, Just return other quiz props.